core/field_image.js

```diff
@@ -116,12 +131,14 @@ Blockly.utils.object.inherits(Blockly.FieldImage, Blockly.Field);
  */
 Blockly.FieldImage.fromJson = function(options) {
   var src = Blockly.utils.replaceMessageReferences(options['src']);
-  var width = Number(Blockly.utils.replaceMessageReferences(options['width']));
-  var height =
-      Number(Blockly.utils.replaceMessageReferences(options['height']));
+  var width = Number(Blockly.utils.replaceMessageReferences(
+      options['width']));
+  var height = Number(Blockly.utils.replaceMessageReferences(
+      options['height']));
   var alt = Blockly.utils.replaceMessageReferences(options['alt']);
   var flipRtl = !!options['flipRtl'];
-  return new Blockly.FieldImage(src, width, height, alt, null, flipRtl);
+  return new Blockly.FieldImage(src, width, height,
+      alt, null, flipRtl, options);
```

Comment:
There's something funny going on here. If you pass in the options property to the constructor you're essentially saying use the new configuration. So we don't "really" need to pass in alt, flipRtl as they're configurable arguments.
Also, interestingly alt here goes through replaceMessageReferences in this path, but it doesn't in the configure_ path, I think that's a bug.

Reply:
> There's something funny going on here. If you pass in the options property to the constructor you're essentially saying use the new configuration. So we don't "really" need to pass in alt, flipRtl as they're configurable arguments.

Yep missed that, I caught it on the number field though so that's good.

>Also, interestingly alt here goes through replaceMessageReferences in this path, but it doesn't in the configure_ path, I think that's a bug.

Good catch, can fix.